### PR TITLE
Use alias analysis to handle internal variables in CB input.

### LINKED ISSUE #7926: Understanding self type in CB
To be done after #7863 

If the editing context `opr` is an access operator with an identifier on the left side, and this identifier is an existing local variable available in our method, then the qualified pattern of the resulting filter should not be defined, and self type should be the type of such local.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #7926

See the new test cases to see what is the improvement.

The self still will be not ideal, but it's because it uses same logic as edges discovery. [When it will be improved](https://github.com/enso-org/enso/pull/8252) this should be improved as well.

https://github.com/enso-org/enso/assets/3919101/25efd56d-5aaa-4f10-957d-8ed6a40f3d9b

### Important Notes

It contains part of #8250. That PR may be reviewed, or here both PRs may be reviewed at once.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  ~~- [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by farmaazon** [comment]
Converted to draft because discovered that this still does not behave ideally

**Comment by farmaazon** [comment]
> Converted to draft because discovered that this still does not behave ideally

The problem is fixed.

File: app/gui2/src/util/containers.ts [hunk 1]
```diff
@@ -80,7 +80,14 @@ export class MappedSet<T extends Object> {
    * The function should return values that are `===`-equal for values that should be considered equal.
    *
    */
-  constructor(private readonly valueMapper: (key: T) => any) {}
+  constructor(
+    private readonly valueMapper: (key: T) => any,
```
Comment (kazcw) [inline app/gui2/src/util/containers.ts:86]: Why is the return type `any` if the docs specify that it should map keys to strings?

**Review by kazcw — APPROVED** [review]

File: app/gui2/src/util/containers.ts (hunk 1 above)
Comment (farmaazon) [reply app/gui2/src/util/containers.ts:86]: It does not have to return string. Updated docs.

**Review by Frizi — APPROVED** [review]

**Pull request merged**